drivers/gles3/storage/material_storage.cpp

```diff
@@ -3068,6 +3084,14 @@ bool SceneShaderData::casts_shadows() const {
 	return !has_alpha || (uses_depth_prepass_alpha && !(depth_draw == DEPTH_DRAW_DISABLED || depth_test == DEPTH_TEST_DISABLED));
 }
 
+bool SceneShaderData::needs_alpha_pass() const {
+	bool has_read_screen_alpha = uses_screen_texture || uses_depth_texture || uses_normal_texture;
+	bool has_base_alpha = (uses_alpha && !uses_alpha_clip) || has_read_screen_alpha;
+	bool has_alpha = has_base_alpha || uses_blend_alpha;
+
+	return has_alpha || depth_draw == SceneShaderData::DEPTH_DRAW_DISABLED || depth_test == SceneShaderData::DEPTH_TEST_DISABLED;
+}
```

Comment:
Based on : https://github.com/godotengine/godot/blob/master/drivers/gles3/rasterizer_scene_gles3.cpp#L200

I could have used my new method in place of the original code, but I didn't want to add regression in a bugfix.